src/Avalonia.X11/X11Window.cs

```diff
@@ -1142,5 +1144,25 @@ public void SetWindowManagerAddShadowHint(bool enabled)
         public AcrylicPlatformCompensationLevels AcrylicCompensationLevels { get; } = new AcrylicPlatformCompensationLevels(1, 0.8, 0.8);
 
         public bool NeedsManagedDecorations => false;
+
+
+        public class SurfacePlatformHandle : IPlatformNativeSurfaceHandle
+        {
+            private readonly X11Window _owner;
+
+            public PixelSize Size => _owner.ToPixelSize(_owner.ClientSize);
+
+            public double Scaling => _owner.RenderScaling;
+
+            public SurfacePlatformHandle(IntPtr handle, string? descriptor, X11Window owner)
```

Comment:
I think the way you've done it for Win32 is better. So just pass X11Window

Reply:
Addressed